### PR TITLE
Precompile the reinitialization folder

### PR DESCRIPTION
**Pull request by peterrum**
references #151

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -107,6 +107,8 @@ if(ADAFLO)
   message("-- Adaflo found: ${ADAFLO}")
   target_compile_definitions(${projectName} PUBLIC MELT_POOL_DG_WITH_ADAFLO)
   target_include_directories(${projectName} PUBLIC ${ADAFLO_INCLUDE})
+  target_compile_definitions(${libName} PUBLIC MELT_POOL_DG_WITH_ADAFLO)
+  target_include_directories(${libName} PUBLIC ${ADAFLO_INCLUDE})
```
Comment (peterrum) [inline CMakeLists.txt:111]: These lines of code needed to be added to the main `CMakeLists.txt` file.

Comment (mschreter) [reply CMakeLists.txt:111]: Thank you Peter, Is this related to the failing adaflo build of #166?

Comment (peterrum) [reply CMakeLists.txt:111]: I am currently rebasing that branch to see if there is any correlation... Not sure yet.

Comment (peterrum) [reply CMakeLists.txt:111]: Jup. This fixes it. Let's get this in as soon as possible and than I'll rebase the branch of the other PR.

Comment (mschreter) [reply CMakeLists.txt:111]: Sounds good. Feel free to merge as as Travis is happy?

File: include/meltpooldg/flow/adaflo_wrapper.hpp [hunk 2]
```diff
@@ -7,11 +7,14 @@
...
 #ifdef MELT_POOL_DG_WITH_ADAFLO
 
+#  include <deal.II/fe/fe_simplex_p.h>
+
 #  include <deal.II/lac/generic_linear_algebra.h>
 
 #  include <meltpooldg/flow/adaflo_wrapper_parameters.hpp>
 #  include <meltpooldg/flow/flow_base.hpp>
 #  include <meltpooldg/interface/scratch_data.hpp>
+#  include <meltpooldg/interface/simulationbase.hpp>
```
Comment (peterrum) [inline include/meltpooldg/flow/adaflo_wrapper.hpp:17]: Just as a not. You will also see that many includes are missing (and the warnings come at places were you don't expect them - my guess we need to clean up the dependencies at one time).

Comment (mschreter) [reply include/meltpooldg/flow/adaflo_wrapper.hpp:17]: Unfortunately, I also observed the same when I introduced the explicit template instantiations :/. Clean-up of includes sounds good. Maybe we can get support from a "cleaning" software there, like https://include-what-you-use.org/.

**Pull request merged**